### PR TITLE
Added missing documentation and constant to PermissionLib

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/src/core/permission/PermissionLib.sol [hunk 1]
```diff
@@ -6,18 +6,35 @@ pragma solidity 0.8.17;
...
+    /// @param Grant The grant operation setting a permission without a condition.
+    /// @param Revoke The revoke operation removing a permission (that was granted with or without a condition).
+    /// @param GrantWithCondition The grant operation setting a permission with a condition.
     enum Operation {
         Grant,
         Revoke,
         GrantWithCondition
     }
 
+    /// @notice A struct containing the information for a permission to be applied on a single target contract without a condation.
```
Comment (novaknole) [inline packages/contracts/src/core/permission/PermissionLib.sol:22]:
```suggestion
    /// @notice A struct containing the information for a permission to be applied on a single target contract without a condition.
```

File: packages/contracts/src/core/permission/PermissionLib.sol [hunk 2]
```diff
@@ -6,18 +6,35 @@ pragma solidity 0.8.17;
 /// @author Aragon Association - 2021-2023
 /// @notice A library containing objects for permission processing.
 library PermissionLib {
+    /// @notice A constant expressing that not condition is applied to a permission.
+    address constant NO_CONDITION = address(0);
```
Comment (novaknole) [inline packages/contracts/src/core/permission/PermissionLib.sol:10]: Doesn't this give a warning ? maybe ambiguity and better add `public` or I completely forgot if you could add `public` at all in the library. Just mentioning. You decide.

File: packages/contracts/src/core/permission/PermissionLib.sol [hunk 3]
```diff
@@ -6,18 +6,35 @@ pragma solidity 0.8.17;
 /// @author Aragon Association - 2021-2023
 /// @notice A library containing objects for permission processing.
 library PermissionLib {
+    /// @notice A constant expressing that not condition is applied to a permission.
```
Comment (novaknole) [inline packages/contracts/src/core/permission/PermissionLib.sol:9]:
```suggestion
    /// @notice A constant expressing that no condition is applied to a permission.
```

File: packages/contracts/src/core/permission/PermissionLib.sol (hunk 2 above)
Comment (Michael-A-Heuer) [reply packages/contracts/src/core/permission/PermissionLib.sol]: Added public a8e74aca1abbbb21bc7fb5ce3cff9d7aca8a83b1

**Review by brickpop — APPROVED** [review]

**Review by brickpop — DISMISSED** [review]

**Review by novaknole — APPROVED** [review]

**Review by brickpop — COMMENTED** [review]
Natspec minor fix

**Review by brickpop — APPROVED** [review]

**Pull request merged**